src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -84,25 +94,26 @@ public void killTask(ExecutorDriver driver, TaskID taskId) {
 
   @Override
   public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
-    log.info("Executor received framework message of length: " + msg.length + " bytes");
+    super.frameworkMessage(driver, msg);
     String messageStr = new String(msg);
+    File nameDir = new File(schedulerConf.getDataDir() + "/name");
     if (messageStr.equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)
         || messageStr.equals(HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE)) {
-      // Initialize the journal node and name node
-      runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
-      // Start the name node
-      startProcess(driver, nameNodeTask);
-      driver.sendStatusUpdate(TaskStatus.newBuilder()
-          .setTaskId(nameNodeTask.taskInfo.getTaskId())
-          .setState(TaskState.TASK_RUNNING)
-          .build());
-      // Start the zkfc node
-      startProcess(driver, zkfcNodeTask);
-      driver.sendStatusUpdate(TaskStatus.newBuilder()
-          .setTaskId(zkfcNodeTask.taskInfo.getTaskId())
-          .setState(TaskState.TASK_RUNNING)
-          .build());
+      if (nameDir.exists()) {
+        log.info(String
+            .format("NameNode data directory %s already exists, not formatting just starting",
+                nameDir));
```

Comment:
"just starting"? But you're not even starting anything anymore. Even if you're recovering a /name metadata dir from a failed over NN, won't you still need to start the NN+ZKFC processes? Please explain